pkg/exclusions/exclusions.go

```diff
@@ -0,0 +1,104 @@
+package exclusions
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/google/osv-scanner/internal/cachedregexp"
+	"github.com/google/osv-scanner/internal/sbom"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// ExcludePackages checks for matches in a Lockfile's Packages against the list of provided exclusion regex patterns.
+// If a match is found, the package is removed from the Lockfile's Packages.
+// The updated Lockfile is returned.
```

Comment:
nit: while I like comments, I feel like a few of them in this file are redundant

e.g. I think it should be pretty obvious given the return type that the updated lockfile is what is being returned, and that a private function called `isRegexValid` checks if strings are valid regexps...

Would you mind doing re-check with that thinking in your head and seeing if there are any comments you're comfortable with dropping? I know this can be very subjective so happy if you come back saying you think we need all these comments 🙂

Reply:
Completely agree. Comments are good, but superfluous commenting can reduce readability in some cases. I do believe it is good practice to leave comments on public functions though.